src/software/ai/hl/stp/tactic/assigned_skill/assigned_skill_tactic.hpp

```diff
@@ -1,87 +1,87 @@
 #pragma once
 
-#include "software/ai/hl/stp/skill/skill_fsm.h"
+#include "software/ai/hl/stp/skill/base_skill.hpp"
 #include "software/ai/hl/stp/tactic/tactic.h"
 
 /**
- * SkillTactic executes primitives generated by a Skill FSM, enabling testing
- * of Skills in simulated gameplay tests.
+ * Implements a Tactic that executes a Skill.
  *
- * To create a SkillTactic, use the AssignedSkillTactic template with the
- * Skill FSM to execute as the template argument.
- */
-class SkillTactic : public Tactic
-{
-   protected:
-    explicit SkillTactic()
-        : Tactic({RobotCapability::Kick, RobotCapability::Chip, RobotCapability::Move})
-    {
-    }
-
-    inline void accept(TacticVisitor& visitor) const override
-    {
-        visitor.visit(*this);
-    }
-};
-
-/**
- * This template lets us generate subtypes of SkillTactic associated
- * with a Skill FSM to execute.
+ * The AssignedSkillTactic can be used in simulated gameplay tests, field tests, or set
+ * plays that require robots to execute a single skill.
  *
- * @tparam TSkillFSM the type of Skill FSM that this tactic will execute
+ * @tparam TSkill the Skill to execute (must inherit BaseSkill)
```

Comment:
can't you do an compile-time inheritance check?

Reply:
I'll try this https://stackoverflow.com/a/73034562  
I tried the using the C++20 concept before but it didn't work with our version of gcc

Reply:
One option I've seen in similar scenarios uses `static_assert` with `std::is_base_on`. e.g. https://gist.github.com/xunkai55/3bea83c38f4f1a52f71f

Reply:
The problem is that `BaseSkill` is a template, so `std::is_base_of` doesn't work. We need to check whether a class derives from any instantiation of `BaseSkill`